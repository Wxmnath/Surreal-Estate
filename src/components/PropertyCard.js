import React from "react";
import PropTypes from "prop-types";
import "../styles/PropertyCard.css";
// import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

function PropertyCard(props) {
  const { title, type, bathrooms, bedrooms, price, city, email } = props;

  return (
    <div className="property-card" data-testid="property-card">
      <div className="property-card__title" data-testid="property-title">
        Title:{title}
      </div>

      <div className="property-card__type" data-testid="property-type">
        {/* <FontAwesomeIcon icon="fa-solid fa-house-chimney-blank" /> */}
        Type:{type}
      </div>

      <div
        className="property-card__bathrooms"
        data-testid="property-bathrooms"
      >
        Bathrooms:{bathrooms}
      </div>

      <div className="property-card__bedrooms" data-testid="property-bedrooms">
        Bedrooms:{bedrooms}
      </div>

      <div className="property-card__price" data-testid="property-price">
        Price:{price}
      </div>

      <div className="property-card__city" data-testid="property-city">
        City:{city}
      </div>

      <div className="property-card__email" data-testid="property-email">
        <a href="mailto:someone@example.com">Email:{email}</a>
      </div>
    </div>
  );
}

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
