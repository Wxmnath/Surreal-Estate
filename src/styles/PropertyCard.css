.property-card {
  display: flex;
  flex-direction: column;
  padding: 15px;
  padding-left: 10px;
  flex-wrap: wrap;
  margin: 10px;
  border: 3px solid #ccc;
  box-sizing: border-box;
  max-width: 100%;
}
