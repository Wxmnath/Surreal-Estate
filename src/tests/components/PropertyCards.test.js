import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "title",
    type: "type",
    bathrooms: 2,
    bedrooms: 3,
    price: 123456,
    city: "Manchester",
    email: "test.email@email.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct values to the props", () => {
    const { getByTestId } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByTestId("property-title")).toHaveClass("property-card__title");
    expect(getByTestId("property-type")).toHaveClass("property-card__type");
    expect(getByTestId("property-bathrooms")).toHaveClass(
      "property-card__bathrooms"
    );
    expect(getByTestId("property-bedrooms")).toHaveClass(
      "property-card__bedrooms"
    );
    expect(getByTestId("property-price")).toHaveClass("property-card__price");
    expect(getByTestId("property-city")).toHaveClass("property-card__city");
    expect(getByTestId("property-email")).toHaveClass("property-card__email");
  });
});
