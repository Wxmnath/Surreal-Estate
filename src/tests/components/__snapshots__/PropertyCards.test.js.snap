// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
    data-testid="property-card"
  >
    <div
      class="property-card__title"
      data-testid="property-title"
    >
      Title:title
    </div>
    <div
      class="property-card__type"
      data-testid="property-type"
    >
      Type:type
    </div>
    <div
      class="property-card__bathrooms"
      data-testid="property-bathrooms"
    >
      Bathrooms:2
    </div>
    <div
      class="property-card__bedrooms"
      data-testid="property-bedrooms"
    >
      Bedrooms:3
    </div>
    <div
      class="property-card__price"
      data-testid="property-price"
    >
      Price:123456
    </div>
    <div
      class="property-card__city"
      data-testid="property-city"
    >
      City:Manchester
    </div>
    <div
      class="property-card__email"
      data-testid="property-email"
    >
      <a
        href="mailto:someone@example.com"
      >
        Email:test.email@email.com
      </a>
    </div>
  </div>
</DocumentFragment>
`;
